ChronosTankDrive/src/main/java/frc/robot/subsystems/shooter/Hood.java

```diff
@@ -113,16 +112,11 @@ public void setHome() {
   }
 
   public void home() {
-    SmartDashboard.putNumber("Homing", 1);
-    m_hoodMotor.set(.2);
-    while (limit.get())
-      ;
-    m_hoodMotor.set(-.25);
-    while (!limit.get())
-      ;
-    m_hoodMotor.set(0);
-    SmartDashboard.putNumber("Homing", 0);
-    setHome();
+    resetHomed();
+    long start=System.currentTimeMillis();
+    while(start+10000>System.currentTimeMillis()&&!homed){
```

Comment:
```suggestion
    long timeout=System.currentTimeMillis() + 10000; // 10000 should be a named constant like TIME_OUT
    while(System.currentTimeMillis()<timeout && !homed){
```
If you are adding values that don't change, then only add them once. It doesn't make a big difference, but every little bit helps and it keeps the code cleaner.